### ISSUE TITLE
tap latency need accelerometer

### ISSUE THREAD

**Issue by AlphaZhang1990** [google/walt#120]
Now tap latency need accelerometer. If the tap speed is different, the result will different. Is there any way to test tap latency not use accelerometer?

**Comment by adlr** [comment]
Sorry, touch up and touch down latency need the accelerometer.

**Comment by kamrik** [comment]
Yes, as Andrew mentioned, tap latency measurement needs the accelerometer. It is used to detect when the probe physically touches the screen. We experimented with other methods, but those were way less accurate.

The result should not depend much on speed, I would expect the variability from speed to be on the order of 1 ms. One way to reduce it is to always do a fairly quick and firm tap.

**Comment by AlphaZhang1990** [comment]
Can I know what is the sample rate of the accelerometer?

**Comment by adlr** [comment]
Mark may know for sure. Looking at the code, the accelerometer is probed in a tight loop without fixed frequency. Thus I would expect it's thousands of times per second. Note that this is an educated guess, not confirmed.

**Comment by kamrik** [comment]
It should be at least several kHz. The accelerometer output is analog and as Andrew mentioned, the microcontroller samples it in a tight loop with no sleep. The ADC on Teensy is known to handle 44.1 kHz audio sampling well, so I would expect it to be in the tens of kHz. But I don't remember ever measuring the actual rate.

**Comment by AlphaZhang1990** [comment]
OK，I will try quick and firm tap to reduce the result change.
